### ISSUE TITLE
flox upgrade is not upgrading packages

### ISSUE THREAD

**Issue by dg-nvm** [flox/flox#1916]
**Describe the bug:**

I issued `flox upgrade` but it does not upgrade packages. At least k9s stays in 0.32.4 when `flox show` shows new version is available

**Steps to reproduce:**

```
λ flox list | grep k9s
k9s: k9s (0.32.4)

λ flox upgrade
ℹ️  No packages need to be upgraded in environment 'alkuzad/default'.

λ flox show k9s | grep 0.32
    k9s@0.32.5
    k9s@0.32.4
```

Result:

No packages need to be upgraded in environment 'alkuzad/default'.

Expected:

k9s bumped to 0.32.5

**Flox Version (run `flox --version` if possible):**

1.2.3

**`uname -a` output:**

Linux VONJLP55M3 6.6.36.3-microsoft-standard-WSL2 #1 SMP PREEMPT_DYNAMIC Sat Jun 29 07:01:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

**Comment by ghudgins** [comment]
@dg-nvm what's in your manifest? the upgrade command will try to fulfill the manifest but it won't change the manifest version requirements

**Comment by dg-nvm** [comment]
@ghudgins 

```
version = 1
#
# This is a Flox environment manifest.
# Visit flox.dev/docs/concepts/manifest/
# or see flox-edit(1), manifest.toml(1) for more information.
#

# List packages you wish to install in your environment inside
# the `[install]` section.
[install]
chezmoi.pkg-path = "chezmoi"
zsh.pkg-path = "zsh"
zsh.version = "=5.9"
deepin-gtk-theme.pkg-path = "deepin.deepin-gtk-theme"
okta-aws-cli.pkg-path = "okta-aws-cli"
yq-go.pkg-path = "yq-go"
yamllint.pkg-path = "yamllint"
yaml2json.pkg-path = "yaml2json"
teleport.pkg-path = "teleport"
ssm-session-manager-plugin.pkg-path = "ssm-session-manager-plugin"
shellcheck.pkg-path = "shellcheck"
moreutils.pkg-path = "moreutils"
kubernetes-helm.pkg-path = "kubernetes-helm"
kubernetes.pkg-path = "kubernetes"
kubectx.pkg-path = "kubectx"
jq.pkg-path = "jq"
jira-cli-go.pkg-path = "jira-cli-go"
jfrog-cli.pkg-path = "jfrog-cli"
hyperfine.pkg-path = "hyperfine"
git-filter-repo.pkg-path = "git-filter-repo"
git.pkg-path = "git"
gimme-aws-creds.pkg-path = "gimme-aws-creds"
gh.pkg-path = "gh"
fzf.pkg-path = "fzf"
fnm.pkg-path = "fnm"
fd.pkg-path = "fd"
direnv.pkg-path = "direnv"
bat.pkg-path = "bat"
antigen.pkg-path = "antigen"
eza.pkg-path = "eza"
neovim.pkg-path = "neovim"
jdk.pkg-path = "jdk"
du-dust.pkg-path = "du-dust"
tldr.pkg-path = "tldr"
glances.pkg-path = "glances"
gtop.pkg-path = "gtop"
dotnet-runtime_8.pkg-path = "dotnet-runtime_8"
nix-tree.pkg-path = "nix-tree"
fdupes.pkg-path = "fdupes"
k9s.pkg-path = "k9s"
wslu.pkg-path = "wslu"

# Set environment variables in the `[vars]` section. These variables may not
# reference once another, and are added to the environment without first
# expanding them. They are available for use in the `[profile]` and `[hook]`
# scripts.
[vars]
_FLOX_ENV_CUDA_DETECTION = "0"
# message = "Howdy"

# The `hook.on-activate` script is run by the *bash* shell immediately upon
# activating an environment, and will not be invoked if Flox detects that the
# environment has previously been activated. Variables set by the script will
# be inherited by `[profile]` scripts defined below. Note that any stdout
# generated by the script will be redirected to stderr.
[hook]
on-activate = """
  #!/bin/bash

  # Not really necessary I believe but I will skip init on VSCODE because 
  # it is unlikely that VSCODE server was started before the shell, which should
  # remove these pythons already
  if [ -n "$RUNNING_IN_VSCODE_SERVER" ];then 
    exit 0
  fi

  mkdir -p ~/.local/bin

  if [ ! -f ~/.local/bin/flox-default-fix.sh ]; then
    cat << 'EOF' > ~/.local/bin/flox-default-fix.sh
 #!/bin/bash

set -eu -o pipefail
 
if [ -z "$FLOX_ENV" ]; then
  echo "$FLOX_ENV not defined, failing removal of python"
  exit 1
fi

if [ -f $FLOX_ENV/bin/python ]; then
  echo "Bogus Python in $FLOX_ENV, removing, needs sudo"
  sudo unlink "$FLOX_ENV/bin/python"
fi

if [ -f $FLOX_ENV/bin/python3 ];then
  echo "Bogus Python3 in $FLOX_ENV, removing, needs sudo"
  sudo unlink "$FLOX_ENV/bin/python3"
fi

EOF

    chmod +x ~/.local/bin/flox-default-fix.sh
  fi
  
  ~/.local/bin/flox-default-fix.sh
"""
# on-activate = """
#     # Set variables, create files and directories
#     venv_dir="$(mktemp -d)"
#     export venv_dir
#
#     # Perform initialization steps, e.g. create a python venv
#     python -m venv "$venv_dir"
#
# """

# Scripts defined in the `[profile]` section are *sourced* by *your shell* and
# inherit environment variables set in the `[vars]` section and by `[hook]` scripts.
# The `profile.common` script is sourced by all shells and special care should be
# taken to ensure compatibility with all shells. The `profile.bash` and `profile.zsh`
# scripts are then sourced by the corresponding shell.
[profile]
# common = """
#     echo "it's gettin' flox in here"
# """
# bash = """
#  """
#zsh = """
#"""

# Additional options can be set in the `[options]` section. Refer to
# manifest.toml(1) for a list of available options.
[options]
cuda-detection=false

# An environment that works on one system is guaranteed to work on the same type
# of system, but other systems may not have the same packages available, etc.
# In order to use the environment on a system you must explicitly add it to the
# `options.systems` list.
systems = ["x86_64-linux"]
```

**Comment by ghudgins** [comment]
as an aside, we know this isn't well explained and need to do some work on the docs and UX to make this more clear. we also might eventually cave and made all installs go to their own groups but that increasing closure size and bites you when you install, say, a library and a runtime binary and need those guarentees...

**Comment by dg-nvm** [comment]
@ghudgins I moved zsh to it's own group but this didn't help, only moving k9s to another one helped. It is a bummer cause I do not want to mark them all specifically in separate group and I have to be up-to-date due to compliance reasons.

**Comment by dcarley** [comment]
Another variation where the now-deprecated `gocode` prevented `go` and others from being upgraded. I had to remove the packages one-by-one to find out why and adding them back one-by-one was made harder by #1670.
```toml
version = 1

[install]
go.pkg-path = "go"
go-task.pkg-path = "go-task"
gotools.pkg-path = "gotools"
gopls.pkg-path = "gopls"
gomodifytags.pkg-path = "gomodifytags"
gotests.pkg-path = "gotests"
gore.pkg-path = "gore"
gocode.pkg-path = "gocode"
```

```
% flox list   
go: go (1.21.5)
go-task: go-task (3.33.1)
gocode: gocode (unstable-2020-04-06)
gomodifytags: gomodifytags (1.16.0)
gopls: gopls (0.14.2)
gore: gore (0.5.7)
gotests: gotests (1.6.0)
gotools: gotools (0.16.1)

% flox upgrade
ℹ️  No packages need to be upgraded in environment 'tmp'.

% flox remove gocode
🗑️  'gocode' uninstalled from environment 'tmp'

% flox upgrade      
⬆️  Upgraded 'go' in environment 'tmp'.
⬆️  Upgraded 'go-task' in environment 'tmp'.
⬆️  Upgraded 'gomodifytags' in environment 'tmp'.
⬆️  Upgraded 'gopls' in environment 'tmp'.
⬆️  Upgraded 'gore' in environment 'tmp'.
⬆️  Upgraded 'gotests' in environment 'tmp'.
⬆️  Upgraded 'gotools' in environment 'tmp'.

% flox list   
go: go (1.22.7)
go-task: go-task (3.38.0)
gomodifytags: gomodifytags (1.17.0)
gopls: gopls (0.16.2)
gore: gore (0.5.7)
gotests: gotests (1.6.0)
gotools: gotools (0.25.0)
```
